test/unit/lib/serverless.test.js

```diff
@@ -174,4 +179,61 @@ describe('test/unit/lib/serverless.test.js', () => {
       expect(serverless.config).to.have.property('servicePath');
     });
   });
+
+  describe('Extend configuration', () => {
+    let awsRegion;
+    const pluginConfig =
+      require('../../fixtures/programmatic/plugin/extend-config-plugin').pluginConfig;
+
+    const serverlessPath = path.resolve(__dirname, '../../../scripts/serverless.js');
+
+    before(() => {
+      awsRegion = process.env.AWS_REGION;
+      process.env.AWS_REGION = 'us-east-1';
+    });
+
+    after(() => {
+      process.env.AWS_REGION = awsRegion;
+    });
```

Comment:
AWS-SDK needs the region to be set. If you did not set it, it will search for credentials file in fixture path and fails.
Perhaps you have something set here. I'll remove it and set it via env.

Reply:
We're working her on provider agnostic feature. We don't have to make any AWS requests here, and it doesn't even have to be AWS provider.